packages/service/src/utils/documents.ts

```diff
@@ -19,15 +19,33 @@ export const createDocument = ({
   doc: Y.Doc;
   user: { id: string };
 }) =>
-  db
-    .insert("document", {
-      value: doc.getText(TEXT_NAME).toJSON(),
-      origin: Buffer.from(Y.encodeStateAsUpdate(doc)),
-      web_rtc_key: "",
-      creator_id: user.id,
-      latest_update_time: new Date(),
-    })
-    .run(pool);
+  db.serializable(pool, async (txnClient) => {
+    try {
+      await db.sql`SAVEPOINT start`.run(txnClient);
+      const dbDoc = await db
+        .insert("document", {
+          value: doc.getText(TEXT_NAME).toJSON(),
+          origin: Buffer.from(Y.encodeStateAsUpdate(doc)),
+          web_rtc_key: "",
+          creator_id: user.id,
+          latest_update_time: new Date(),
+        })
+        .run(txnClient);
+      return {
+        document: dbDoc,
+        user_document: await db
+          .insert("user_document", {
+            document_id: dbDoc.id,
+            user_id: user.id,
+          })
```

Comment:
we shouldn't need to do this when creating a document anymore. We should be able to keep the `user_document` mutations isolated on the invite/remove user routes